### PR TITLE
split db logic from settings manager and test them separately

### PR DESCRIPTION
**Pull request by icewind1991**
Mocking the db/query builder is a pita, to much work and generally useless.

By splitting of the database parts from the manager we can test those parts against a real db and improve the tests for the rest

### CODE REVIEW

**Review by rullzer — APPROVED** [review]

File: lib/private/Settings/Manager.php [hunk 1]
```diff
@@ -329,24 +260,17 @@ private function query($className) {
 	public function getAdminSections() {
 		// built-in sections
 		$sections = [
-			 0 => [new Section('server',        $this->l->t('Server settings'), 0)],
-			 5 => [new Section('sharing',       $this->l->t('Sharing'), 0)],
-			45 => [new Section('encryption',    $this->l->t('Encryption'), 0)],
-			98 => [new Section('additional',    $this->l->t('Additional settings'), 0)],
-			99 => [new Section('tips-tricks',   $this->l->t('Tips & tricks'), 0)],
+			0 => [new Section('server', $this->l->t('Server settings'), 0)],
```
Comment (blizzz) [inline lib/private/Settings/Manager.php]: less readable

File: lib/private/Settings/Mapper.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	private $dbc;
+
+	/**
+	 * @param IDBConnection $dbc
+	 */
+	public function __construct(IDBConnection $dbc) {
+		$this->dbc = $dbc;
+	}
+
+	public function getAdminSettingsFromDB($section) {
```
Comment (blizzz) [inline lib/private/Settings/Mapper.php]: PHP Doc missing

File: lib/private/Settings/Mapper.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	 */
+	public function remove($table, $className) {
+		$query = $this->dbc->getQueryBuilder();
+		$query->delete($table)
+			->where($query->expr()->eq('class', $query->createNamedParameter($className)));
+
+		$query->execute();
+	}
+
+	public function update($table, $idCol, $id, $values) {
```
Comment (blizzz) [inline lib/private/Settings/Mapper.php]: PHP Doc missing

**Review by blizzz — COMMENTED** [review]
Missing PHP Doc, looks good otherwise

**Comment by MorrisJobke** [comment]
@icewind1991 Please add the PHPDoc, thanks.

**Comment by icewind1991** [comment]
> Please add the PHPDoc, thanks.

Done

**Review by MorrisJobke — APPROVED** [review]
Makes sense and still works 👍

**Pull request merged**